core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -168,6 +236,50 @@ protected void doStop() throws Exception {
     return server;
   }
 
+  private static String[] getList(String original) {
+    String[] parts = original.split(",");
+    for (int i = 0; i < parts.length; i++) {
+      parts[i] = parts[i].trim();
+    }
+    return parts;
+  }
+
+  // TODO: delete deprecatedPort parameter when `PORT_CONFIG` is deprecated. It's only used to support the deprecated
+  //       configuration.
+  static List<URI> parseListeners(String listenersStr, int deprecatedPort) {
+    String[] parts = getList(listenersStr);
+
+    // handle deprecated case, using PORT_CONFIG.
+    // TODO: remove this when `PORT_CONFIG` is deprecated, because LISTENER_CONFIG will have a default value which
+    //       includes the default port.
+    if (parts.length == 0 || parts[0].isEmpty()) {
+      parts = new String[] {"http://0.0.0.0:" + deprecatedPort};
```

Comment:
In Kafka, we bind in the following way:

```scala
val socketAddress =
      if(host == null || host.trim.isEmpty)
        new InetSocketAddress(port)
      else
        new InetSocketAddress(host, port)
    val serverChannel = ServerSocketChannel.open()
    serverChannel.configureBlocking(false)
    serverChannel.socket().setReceiveBufferSize(recvBufferSize)
    try {
      serverChannel.socket.bind(socketAddress)
```

If the host is not present, we basically end up using:

```java
    /*
     * Returns the InetAddress representing anyLocalAddress
     * (typically 0.0.0.0 or ::0)
     */
    static InetAddress anyLocalAddress() {
        return impl.anyLocalAddress();
    }
```

When it comes to what we register to ZK though, we do:

```scala
if (endpoint.host == null || endpoint.host.trim.isEmpty)
  EndPoint(InetAddress.getLocalHost.getCanonicalHostName, endpoint.port, endpoint.protocolType)
```

So we use the `